src/redprl/refiner.fun

```diff
@@ -17,7 +17,7 @@ struct
 
   structure Hyp =
   struct
-    fun Project z alpha jdg =
+    fun Project z _ jdg =
```

Comment:
Maybe I can add that I did not blindly delete all unused code---you can see the build log. The `alpha` is an annoying case because it clutters the log when I tried to carefully review every case, and I have not came up with a good combinator or style to both problems. Could we have another wrapper for the "pure" rules that do not use the name generator? The `makeGoal'` etc were created to solve readability and warnings at the same time.